src/framework/plugin.ts

```diff
@@ -132,9 +133,18 @@ export type RuntimeContributions<C extends {} = any> = {
 
 type RuntimePlugin = () => RuntimeContributions
 
+type DeepPartial<T extends Record<string, any>> = {
+  [P in keyof T]?: T[P] extends Record<string, any> ? DeepPartial<T[P]> : T[P]
+} & { [x: string]: any }
+
+export type TestingContributions = DeepPartial<TestContext>
+
+type TestingPlugin = () => TestingContributions
+
 export type Lens = {
   runtime: CallbackRegistrer<RuntimePlugin>
   workflow: CallbackRegistrer<WorkflowDefiner>
+  testing: CallbackRegistrer<TestingPlugin>
```

Comment:
Diverging here between how context types and test types works. With context types plugins are given an api to provide typings. Then the framework takes care of integrating that into the global context typings of nexus.

Right now our approach with test context typing is that plugin will handle extending the type itself. There is an advantage to this approach (assuming post-install): it leads to correct typings after plugin installation, without having to run any `santa` command.

I would like to see the plugin interface become uniform in the future though. Maybe the context type strategy should follow what we're doing for test context type.

Reply:
https://github.com/prisma-labs/graphql-santa/issues/301